lib/slax/poker.ex

```diff
@@ -63,15 +63,41 @@ defmodule Slax.Poker do
 
   def decide(round, score) do
     {org, repo, issue} = Github.parse_repo_org_issue(round.issue)
-
     client = Tentacat.Client.new(%{access_token: Github.api_token()})
 
-    case Tentacat.Issues.update(client, org, repo, issue, %{labels: ["Points: #{score}"]}) do
-      {200, _issue, _http_response} ->
-        :ok
-
+    with {200, labels, _http_response} <- Tentacat.Issues.Labels.list(client, org, repo, issue),
+         points_label <- find_points_label(labels),
+         :ok <- maybe_remove_label(client, org, repo, issue, points_label),
+         {200, _issue, _http_response} <-
+           Tentacat.Issues.Labels.add(client, org, repo, issue, ["Points: #{score}"]) do
+      :ok
+    else
       {_response_code, %{"message" => error_message}, _http_response} ->
         {:error, error_message}
     end
   end
+
+  defp find_points_label(labels) do
+    with %{"name" => label} <-
+           Enum.find(labels, fn
+             %{"name" => "Points: " <> _number} -> true
+             %{"name" => _} -> false
+           end) do
+      label
+    else
+      nil ->
+        nil
+    end
```

Comment:
One very general rule of thumb: `with` is best reserved for things that can fail. Other things like `if`, `case`, and `cond` are better for control flow, in that order of preference usually (using the simplest one that works). In this case, I think it's more readable like this

```suggestion
    points_label = Enum.find(labels, fn
      %{"name" => "Points: " <> _number} -> true
      _ -> false
    end)
    
    if points_label, do: points_label["name"], else: nil
```